### ISSUE TITLE
[Core Mechanic] Threat System

### ISSUE THREAD

**Issue by Databasegm** [FeenixServerProject/Archangel_2.4.3_Bugtracker#1893]
From my testing with Defiance, which was bugged, and now appears to be fixed on the PTR, I noticed some issues with threat and aggro.  I'll list my test results here, and do some more testing in the future with different classes.
- While attacking with a fresh 70 (no talents) Warrior in melee range, I did 10000 damage in auto attacks (aka 10000 threat). I pulled aggro with a Hunter attacking at 20 yards range at 10000 damage.
- With a Rogue I did 10000 threat in melee, and then pulled aggro with a paladin (No talents) in melee range with around 6000 threat.

**To recap how threat and aggro should work:**
1) For an NPC to change targets, a player in melee range (5 yards or less) needs to have more that 110% of the current target's threat
2) For an NPC to change targets, a player outside of melee range (Over 5 yards) needs to have more that 130% of the current target's threat.

Feel free to contact me via PM to discuss further.
